pkg/remote/client_test.go

```diff
@@ -0,0 +1,103 @@
+package remote
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	prompb "go.buf.build/grpc/go/prometheus/prometheus"
+)
+
+func TestNewWrtiteClient(t *testing.T) {
+	t.Parallel()
+	t.Run("DefaultConfig", func(t *testing.T) {
+		t.Parallel()
+		wc, err := NewWriteClient("http://example.com/api/v1/write", nil)
+		require.NoError(t, err)
+		require.NotNil(t, wc)
+		assert.Equal(t, wc.cfg, &HTTPConfig{})
+	})
+
+	t.Run("CustomConfig", func(t *testing.T) {
+		t.Parallel()
+		hc := &HTTPConfig{Timeout: time.Second}
+		wc, err := NewWriteClient("http://example.com/api/v1/write", hc)
+		require.NoError(t, err)
+		require.NotNil(t, wc)
+		assert.Equal(t, wc.cfg, hc)
+	})
+
+	t.Run("InvalidURL", func(t *testing.T) {
+		t.Parallel()
+		wc, err := NewWriteClient("fake://bad url", nil)
+		require.Error(t, err)
+		assert.Nil(t, wc)
+	})
+}
+
+func TestClientStore(t *testing.T) {
+	t.Parallel()
+	h := func(rw http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, r.Header.Get("Content-Encoding"), "snappy")
+		assert.Equal(t, r.Header.Get("Content-Type"), "application/x-protobuf")
+		assert.Equal(t, r.Header.Get("User-Agent"), "k6-prometheus-rw-output")
+		assert.Equal(t, r.Header.Get("X-Prometheus-Remote-Write-Version"), "0.1.0")
```

Comment:
It's probably okay to test that this header isn't empty. Otherwise, a test should be adjusted every release.

Reply:
They are defined by the Promethus specs, I added a link to it

Reply:
Yeah, my fault. For some reason, I thought that this is about an extension :facepalm: